lib/ansible/modules/notification/cisco_spark.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'status': ['stableinterface'],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: cisco_spark
+short_description: Send a message to a Cisco Spark Room or Individual
+description:
+    - Utilize the Cisco Spark API (https://developer.ciscospark.com/getting-started.html) to send a message to a Spark Room or Individual in either text or markdown format.
+version_added: "1.0"
+author: Drew Rusell (@drusse11)
+notes:
+  - The receipient_id type must be valid for the supplied recipient_type.
+  - Full API documentation can be found at https://developer.ciscospark.com/endpoint-messages-post.html
+
+options:
+
+  recipient_type:
+    description:
+       - The request parametmer you would like to send the message to.
+       - Messages can be sent to either a room or individual (by ID or E-Mail).
+    required: True
+    choices: ['roomId', 'toPersonEmail', 'toPersonId']
+
+  recipient_id:
+    description:
+      - The unique identifier associated with the supplied recipient_type
+    required: true
+
+  message_type:
+    description:
+       - Specifies how you would like the message formatted.
+    required: False
```

Comment:
Looking at the example documentation they show `required: true or false` so I would argue that having `required: False` is best practice. I am also a believe in `Explicit is better than implicit` and prefer to keep this.